src/modules/indexer/client.ts

```diff
@@ -65,14 +65,20 @@ export const IndexerSDK = {
 
     const json = await res.json();
 
-    return sortWith(
-      [
-        //@ts-ignore   
-        ascend(prop('instantSalePrice')),
-        descend(prop('highestBid')),
-        ascend(prop('endsAt'))
-      ],
-      json.result
-    )
+    return compose<any, Listing[]>(
+      sortWith<Listing>(
+        [
+          ascend<any>(prop('instantSalePrice')),
+          descend(prop('highestBid')),
+          ascend(prop('endsAt'))
+        ]
+      ),
+      when<Listing, any>(
+        always(not(isNil(approved))),
+        //@ts-ignore
+        filter(pipe(prop('subdomain'), (subdomain) => includes(subdomain, approved)))
+      )
+    // @ts-ignore
+    )(json.result);
```

Comment:
Nearly 60% of the time spent on types and still didn't get it. I will master your ramda typing. I promise you.

Reply:
lol

Reply:
go back to rust